terraform/install.sh

```diff
@@ -37,25 +39,93 @@ getBillingAccount() {
     log "To list active billing accounts, run:"
     log "gcloud beta billing accounts list --filter open=true"
     exit 1;
-  elif [[ $(echo "${found_accounts}" | wc -l) -gt 1 ]]; then
+  fi
+
+  # store (name:id) info in a map
+  IFS_bak=$IFS
+  declare -A map
+  acc_ids=$(gcloud beta billing accounts list --format="value(displayName,name)" --filter open=true)
+  IFS=$'\n'
+  acc_ids=($acc_ids)
+  for acc in ${acc_ids[@]}
+  do
+    IFS=$'\t'
+    acc=($acc)
+    IFS=$'\n'
+    map[${acc[0]}]=${acc[1]}
+  done
+
+  if [[ $(echo "${found_accounts}" | wc -l) -gt 1 ]]; then
       log "Which billing account would you like to use?:"
-      IFS_bak=$IFS
       IFS=$'\n'
       select opt in ${found_accounts} "cancel"; do
         if [[ "${opt}" == "cancel" ]]; then
           exit 0
         elif [[ -z "${opt}" ]]; then
           log "invalid response"
         else
-          billing_acct=$opt
+          billing_acct=${opt}
+          billing_id=${map[$billing_acct]}
```

Comment:
will this always find an item in the map?

Reply:
It should be.